climpred/bootstrap.py

```diff
@@ -451,9 +451,14 @@ def bootstrap_compute(
         hind,
         metric=metric,
         comparison=comparison,
+        dim=dim,
         function_name=inspect.stack()[0][3],  # take function.__name__
         metadata_dict=metadata_dict,
     )
+    # Ensure that the lead units get carried along for the calculation. The attribute
+    # tends to get dropped along the way due to ``xarray`` functionality.
```

Comment:
I think this is caused in `compute_hindcast` because we loop over `lead`. then the `lead` in skill is from `xr.concat(list,'lead')` and that `lead` is newly created / unrelated to the input `lead`

Reply:
Got it. This could be handled in the future with a decorator or somewhere internally in the compute functions but this is fine for now.